pmacApp/src/pmacController.cpp

```diff
@@ -4189,9 +4161,128 @@ asynStatus pmacController::readDeviceType() {
     }
   }
   debug(DEBUG_VARIABLE, functionName, "Read device CPU", cpu_);
+
   return status;
 }
 
+asynStatus pmacController::getCpuNumCores() {
+  asynStatus status = asynSuccess;
+  static const char *functionName = "getCpuNumCores";
+
+  debug(DEBUG_FLOW, functionName);
+
+  if (strcmp(cpu_.c_str(),"") == 0) {
+    status = asynError;
+    debug(DEBUG_ERROR, functionName, "Error reading card cpu");
+  }
+
+  if (status == asynSuccess) {
+    // Single-core
+    if (strcmp(cpu_.c_str(),"PowerPC,460EX") == 0) {
+      cpuNumCores_ = 1;
+    // Dual-core
+    } else if (strcmp(cpu_.c_str(),"PowerPC,APM86xxx") == 0 ||
+               strcmp(cpu_.c_str(),"arm,LS1021A") == 0) {
+      cpuNumCores_ = 2;
+    // Quad-core
+    } else if (strcmp(cpu_.c_str(),"arm,LS1043A") == 0) {
+      cpuNumCores_ = 4;
+    } else {
+      status = asynError;
+      debug(DEBUG_ERROR, functionName,
+            "Error determining the number of cores from CPU type");
+    }
+  }
+
+  if(status == asynSuccess) {
+    if(cpuNumCores_ < 1 || cpuNumCores_ > PPMAC_CPU_MAXCORES) {
+      debugf(DEBUG_ERROR, functionName,
+            "Detected number of CPU Cores (%d) is out of supported range (1 to 4)", cpuNumCores_);
```

Comment:
This string can use the PPMAC_CPU_MAXCORES it is evaluating against instead of '4'